pkg/worker/client.go

```diff
@@ -51,12 +53,27 @@ type ScrapeDump struct {
 	Dumper  dumper.Dumper
 }
 
-func New(pollTime time.Duration, logger *zap.Logger, workList WorkGetter) ScrapeAndDumpClient {
-	return ScrapeAndDumpClient{
-		workList,
-		pollTime,
-		logger,
+type Option = func(*ScrapeAndDumpClient)
+
+func WithCircuitBreaker(c *circuitbreaker.CircuitBreaker) func(*ScrapeAndDumpClient) {
+	return func(x *ScrapeAndDumpClient) {
+		x.cb = c
+	}
+}
+
+func New(pollTime time.Duration, logger *zap.Logger, workList WorkGetter, opts ...Option) ScrapeAndDumpClient {
```

Comment:
I dig this a lot! I soured on functional arguments because I saw a lot of people trying to cram everything into them and frankly positional arguments have a lot of advantages, so I really like the required-fields-are-positional and everything-optional-is-functional approach